### PR TITLE
Fix: Clear All Filters Badge Visibility in Resource page

### LINKED ISSUE #9556: Clear all filter badge is visible even after all filters are cleared in Resource page
**Describe the bug**
The "Clear all filters" badge remains visible even after all filters have been cleared on the resource page.

**To Reproduce**

1. Go to the Resource tab.
2. Apply any filter(s) using the available options.
3. Remove all filters by clicking the (X) on the individual filter badges.
4. Observe that the "Clear all filters" badge is still visible despite no filters being applied.

**Expected behavior**
The "Clear all filters" badge should disappear when no filters are active, ensuring consistency with the filter status.

**Screenshots**

https://github.com/user-attachments/assets/4f8c3ba6-0995-4d98-8ff7-e2f41994b24c

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by Mahendar0701**
## Proposed Changes

- Fixes #9556 
- Ensures that the "Clear all filters" badge is visible only when filters are selected.
- Removed the default emergency=-- value being sent as part of the filters, which was being applied unnecessarily.

https://github.com/user-attachments/assets/18282e6c-660f-4e81-84cd-9f3634fd87f4

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced querying mechanism for fetching facility data in the Resource Filter component.
	- Introduced a new method for date filtering within the filter utility.

- **Bug Fixes**
	- Adjusted default value for the emergency filter from `"--"` to an empty string `""`.

### CODE REVIEW

File: src/components/Resource/ResourceFilter.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { RESOURCE_FILTER_ORDER } from "@/common/constants";
 import { RESOURCE_CHOICES } from "@/common/constants";
 
 import routes from "@/Utils/request/api";
-import useTanStackQueryInstead from "@/Utils/request/useQuery";
+import useQuery from "@/Utils/request/useQuery";
```
Comment (rithviknishad) [inline src/components/Resource/ResourceFilter.tsx]: this is just renaming the import. use useQuery from tanstack query instead.

https://docs.ohc.network/docs/care/development/data-fetching-in-care/

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@Mahendar0701 the entire resource filters are not working, make sure its covered in the issue

File: src/components/Resource/ResourceFilter.tsx (hunk 1 above)
Comment (Mahendar0701) [reply src/components/Resource/ResourceFilter.tsx]: @rithviknishad  did we implemented this anywhere in care_fe for reference ?

In the current version of tanstack the `onSuccess`, `onError`, and `onSettled `callbacks have been removed from the useQuery options. I got to know that these callbacks can now be handled using `useEffect` or other hooks.

is below approach acceptable?

```js
  const { isLoading: orginFacilityLoading, data } = useQuery({
    queryKey: [routes.getAnyFacility.path, filter.origin_facility],
    queryFn: () =>
      query(routes.getAnyFacility, {
        pathParams: { id: filter.origin_facility },
      }),
    enabled: filter.origin_facility !== undefined,
  });

  useEffect(() => {
    if (data) {
      setFilterState({
        origin_facility_ref: filter.origin_facility === "" ? "" : data,
      });
    }
  }, [data]);

Comment (Jacobjeevan) [reply src/components/Resource/ResourceFilter.tsx]: Yea, that approach works.

File: src/components/Resource/ResourceFilter.tsx [hunk 2]
```diff
@@ -40,51 +42,59 @@ export default function ListFilter(props: any) {
...
-          setFilterState({
-            origin_facility_ref: filter.origin_facility === "" ? "" : data,
-          });
-        }
-      },
-    },
-  );
+  const { isLoading: orginFacilityLoading, data: originFacilityData } =
+    useQuery({
+      queryKey: [routes.getAnyFacility.path, filter.origin_facility],
```
Comment (Jacobjeevan) [inline src/components/Resource/ResourceFilter.tsx]:
```suggestion
      queryKey: ["getAnyFacility", filter.origin_facility],
```

Same goes for ones below 👍

**Review by Jacobjeevan — COMMENTED** [review]
Other than that, lgtm

**Comment by Jacobjeevan** [comment]
No longer relevant.

**Pull request closed without merging**